markdown.go

```diff
@@ -241,6 +241,16 @@ func (p *parser) getRef(refid string) (ref *reference, found bool) {
 	return ref, found
 }
 
+func (p *parser) isFootnote(ref *reference) bool {
+	for _, v := range p.notes {
```

Comment:
You are right. I thought the two string cnversions may cost too much. A map is a good way to deal this problem. I will try to do this.